src/Blockcore.Indexer.Core/Storage/Mongo/MongoData.cs

```diff
@@ -830,39 +831,21 @@ private class MapAddressBag
          public bool CoinBase;
          public bool CoinStake;
 
-         public List<InputTable> Inputs = new List<InputTable>();
-         public List<OutputTable> Ouputs = new List<OutputTable>();
+         public List<InputTable> Inputs = new();
+         public List<OutputTable> Ouputs = new();
       }
 
       public async Task DeleteBlockAsync(string blockHash)
       {
-         SyncBlockInfo block = BlockByHash(blockHash);
-
-         FilterDefinition<InputTable> addrForInputFilter = Builders<InputTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> input = InputTable.DeleteManyAsync(addrForInputFilter);
-
-         FilterDefinition<OutputTable> addrForOutputFilter = Builders<OutputTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> output = OutputTable.DeleteManyAsync(addrForOutputFilter);
-
-         // delete the transaction
-         FilterDefinition<TransactionBlockTable> transactionFilter = Builders<TransactionBlockTable>.Filter.Eq(info => info.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> transactions = TransactionBlockTable.DeleteManyAsync(transactionFilter);
-
-         // delete computed
-         FilterDefinition<AddressComputedTable> addrCompFilter = Builders<AddressComputedTable>.Filter.Eq(addr => addr.ComputedBlockIndex, block.BlockIndex);
-         Task<DeleteResult> addressComputed = AddressComputedTable.DeleteManyAsync(addrCompFilter);
-
-         // delete computed history
-         FilterDefinition<AddressHistoryComputedTable> addrCompHistFilter = Builders<AddressHistoryComputedTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> addressHistoryComputed = AddressHistoryComputedTable.DeleteManyAsync(addrCompHistFilter);
+          SyncBlockInfo block = BlockByHash(blockHash);
 
-         // delete computed utxo
-         FilterDefinition<AddressUtxoComputedTable> addrCompUtxoFilter = Builders<AddressUtxoComputedTable>.Filter.Eq(addr => addr.BlockIndex, block.BlockIndex);
-         Task<DeleteResult> addressUtxoComputed = AddressUtxoComputedTable.DeleteManyAsync(addrCompUtxoFilter);
+          var rewindTask = globalState.IndexModeCompleted
```

Comment:
I don't think `IndexModeCompleted` is the right flag to use here becuase its only set after the indexer task is run, so if on startup we have uncomplete blocks that will trigger a rewind even if indexes are built.

I think we need to check in mongo if indexes are built instead, or a combination of both options 

```
if(!globalState.IndexModeCompleted)
{
  // call mongo to check if indexes are built.
}
```

Reply:
Don't we have a way to indicate to all tasks that the state has chagned from IDB?